packages/core/src/browser/window/default-window-service.spec.ts

```diff
@@ -0,0 +1,58 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Container } from 'inversify';
+import { ContributionProvider } from '../../common';
+import { CorePreferences } from '../core-preferences';
+import { FrontendApplicationContribution } from '../frontend-application';
+import { DefaultWindowService } from './default-window-service';
+import assert = require('assert');
+
+describe('DefaultWindowService', () => {
+    it('onWillStop should be called on every contribution', () => {
+        class TestFrontendApplicationContribution implements FrontendApplicationContribution {
+            constructor(private preventUnload: boolean) { }
+            onWillStopCalled = false;
+            onWillStop(): boolean {
+                this.onWillStopCalled = true;
+                return this.preventUnload;
+            }
+        }
+        const container = new Container();
+        container.bind(DefaultWindowService).toSelf().inSingletonScope();
+        const frontendContributions: TestFrontendApplicationContribution[] = [
+            // random contributions
+            new TestFrontendApplicationContribution(true),
+            new TestFrontendApplicationContribution(false),
+            new TestFrontendApplicationContribution(false),
+            new TestFrontendApplicationContribution(true),
+            new TestFrontendApplicationContribution(false),
+        ];
+        container.bind<Partial<ContributionProvider<FrontendApplicationContribution>>>(ContributionProvider)
+            .toConstantValue({
+                getContributions: () => frontendContributions,
+            })
+            .whenTargetNamed(FrontendApplicationContribution);
+        container.bind<Partial<CorePreferences>>(CorePreferences)
+            .toConstantValue({
+                'application.confirmExit': 'always',
```

Comment:
The test would be more compelling if this weren't `always`, since if it's `always`, `canUnload` will try to prevent unload. If the preference is `ifRequired` and `canUnload` returns false, then we know that some contribution prevented the unloading, but all contributions were still called. Still, it's a minor nit, and I'm satisfied as is.

Reply:
Good catch, I improved the tests a little bit.